Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs

```diff
@@ -102,13 +111,31 @@ protected RelationshipsAuthorizationStrategyBase()
                 Filters = filters,
                 Operator = FilterOperator.Or
             };
-            
+
+            string GetNonRoleNamedProperty(string propertyName)
+            {
+                //Look for the corresponding non - role named property on the model using its DefiningConcreteProperty
+
+                var schema = authorizationContext.Type.GetCustomAttribute<SchemaAttribute>(false)?.Schema;
```

Comment:
This code is going to be executed _a lot_, so we need to try to make sure we are doing it in a performant way. Reflection is slow, so I think that we should introduce a private `ConcurrentDictionary` and the use of the `GetOrAdd` method here to remember the answers and avoid the reflection hit on every authorization call. The answers are also not specific to each entity, so that logic is valuable for first determining the "originating" name the first time, but no need to incorporate that into the key -- just use the (possibly) role-named property as the key, and the resolved non-role-named property as the value.

Ultimately, we only have about 10 total property name entries to resolve... Student/Staff/Parent USIs, and the EdOrgId, and the EdOrg subtypes, and the occasional role-named EdOrgId-derived values (e.g. ParentEducationOrganizationId, and soon, the ResponsibilitySchoolId in DisciplineIncident).

Reply:
Added a cache around it